### PR TITLE
Update IL.targets to handle more properties

### PR DESCRIPTION
**Pull request by richardlford**
Some of the ilasm tests we are porting require setting ilasm switches.
This change adds new properties that if provided set ilasm switches.

If FoldIdenticalMethods is True, provide /fold
If SizeOfStackReserve is not '', provide /stack=value
These DebugType values are recognized:

- Full => /DEBUG
- Impl => /DEBUG=IMPL
- PdbOnly => /DEBUG=OPT
- Any other value, including "" or "NONE" gives no /DEBUG switch

If Optimize = 'True', provide /OPTIMIZE

### CODE REVIEW

**Comment by richardlford** [comment]
@pgavlin, PTAL.

File: tests/src/IL.targets [hunk 1]
```diff
@@ -13,11 +13,17 @@
     <PropertyGroup>
       <_OutputTypeArgument Condition="'$(OutputType)' == 'Library'">/DLL</_OutputTypeArgument>
       <_OutputTypeArgument Condition="'$(OutputType)' == 'Exe'">/EXE</_OutputTypeArgument>
-
-      <_KeyFileArgument Condition="'$(KeyOriginatorFile)' != ''">/KEY=$(KeyOriginatorFile)</_KeyFileArgument>
+      <_IlasmSwitches>/QUIET /NOLOGO</IlasmSwitches>
```
Comment (pgavlin) [inline tests/src/IL.targets]: Looks like the closing tag here is wrong (should be `</_IlasmSwitches>`).

**Comment by richardlford** [comment]
@pgavlin, I've pushed a fix for the closing tag.

**Comment by pgavlin** [comment]
Looks like `TailcallVerifyWithPrefix` is fails due to an assertion change ([log](http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/debug_windows_nt_prtest/1171/testReport/junit/JIT/_opt_Tailcall_TailcallVerifyWithPrefix_TailcallVerifyWithPrefix_/_opt_Tailcall_TailcallVerifyWithPrefix_TailcallVerifyWithPrefix_cmd/)). Please add an appropriate exclusion.

Not sure about the other tests (esp. the Linux failures), as I cannot repro them locally.

**Comment by richardlford** [comment]
Pushed exclusions for failing tests, squashed and rebased.

**Comment by richardlford** [comment]
I pushed a correction for two of the Windows exclusions.
Since non-windows exclusions require a full path I've been using
a full path for Windows exclusions also. But I accidentally left
the "/*" at the end of two of the exclusions.

**Comment by richardlford** [comment]
Pushed one more exclusion for a test that only failed on linux.

**Comment by pgavlin** [comment]
LGTM.

**Pull request merged**